windows/RNPrint/RNPrint.cpp

```diff
@@ -321,53 +321,55 @@ namespace winrt::RNPrint
 
     printOptions.isLandscape = (options.find("isLandscape") != options.end() ? options["isLandscape"].AsBoolean() : false);
     printOptions.jobName = (options.find("jobName") != options.end() ? options["jobName"].AsString() : defaultJobName);
-
-    xaml::FrameworkElement root{ nullptr };
-
-    auto window = xaml::Window::Current();
-
-    if (window != nullptr)
-    {
-      root = window.Content().as<xaml::FrameworkElement>();
-    } else
-    {
-      if (auto xamlRoot = React::XamlUIService::GetXamlRoot(reactContext.Properties().Handle()))
+    reactContext.UIDispatcher().Post([=]()
       {
-        root = xamlRoot.Content().as<xaml::FrameworkElement>();
-      }
-    }
+        xaml::FrameworkElement root{ nullptr };
 
-    if (!root)
-    {
-      cleanUp();
-      promise.Reject("A valid XAML root was not found.");
-      return;
-    }
+        auto window = xaml::Window::Current();
 
-    printCanvas = searchForPrintCanvas(root);
-
-    if (!printCanvas)
-    {
-      cleanUp();
-      promise.Reject("The XAML Canvas named \"RNPrintCanvas\" was not found.");
-      return;
-    }
+        if (window != nullptr)
+        {
+          root = window.Content().as<xaml::FrameworkElement>();
+        } else
+        {
+          if (auto xamlRoot = React::XamlUIService::GetXamlRoot(reactContext.Properties().Handle()))
+          {
+            root = xamlRoot.Content().as<xaml::FrameworkElement>();
+          }
```

Comment:
You want to use try_as here instead of as since as well crash if it isn't a framework element whereas try_as will return null

Reply:
Thank you the review. Sounds good. I've push that change.